network/src/worker.rs

```diff
@@ -167,3 +169,84 @@ impl WorkerNetwork {
         handlers
     }
 }
+
+pub struct WorkerToPrimaryNetwork {
+    address: Option<Multiaddr>,
+    client: Option<WorkerToPrimaryClient<Channel>>,
+    config: mysten_network::config::Config,
+    retry_config: RetryConfig,
+    executor: BoundedExecutor,
+}
+
+impl Default for WorkerToPrimaryNetwork {
+    fn default() -> Self {
+        let retry_config = RetryConfig {
+            // Retry for forever
+            retrying_max_elapsed_time: None,
+            ..Default::default()
+        };
+
+        Self {
+            address: None,
+            client: Default::default(),
+            config: Default::default(),
+            retry_config,
+            executor: BoundedExecutor::new(MAX_TASK_CONCURRENCY, Handle::current()),
+        }
+    }
+}
+
+impl WorkerToPrimaryNetwork {
+    pub async fn send<PublicKey: VerifyingKey>(
```

Comment:
Please reproduce & adapt the "Safety" comment you'll find on send above. Ideally, we would be able to justify of a bound on the # of messages sent per round.

Reply:
fixed